File: typescript-client-sdk/src/sdk/models/operations/puteventsdeprecateeventid.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { Expose, Type } from "class-transformer";


export class PutEventsDeprecateEventIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=event_id" })
  eventId: string;
}

export class PutEventsDeprecateEventIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: PutEventsDeprecateEventIdPathParams;
}

export class PutEventsDeprecateEventId400ApplicationJSON extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "status" })
  status?: number;

  @SpeakeasyMetadata()
  @Expose({ name: "title" })
  title?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "type" })
  type?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "validation_errors" })
  validationErrors?: string[];
}

export class PutEventsDeprecateEventId200ApplicationJSON extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "deprecated" })
  deprecated?: string;
}

export class PutEventsDeprecateEventIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  putEventsDeprecateEventId200ApplicationJSONObject?: PutEventsDeprecateEventId200ApplicationJSON;

  @SpeakeasyMetadata()
  putEventsDeprecateEventId400ApplicationJSONObject?: PutEventsDeprecateEventId400ApplicationJSON;
}

File: go-client-sdk/pkg/models/operations/puteventsdeprecateeventid.go
package operations

type PutEventsDeprecateEventIDPathParams struct {
	EventID string `pathParam:"style=simple,explode=false,name=event_id"`
}

type PutEventsDeprecateEventIDRequest struct {
	PathParams PutEventsDeprecateEventIDPathParams
}

type PutEventsDeprecateEventID400ApplicationJSON struct {
	Status           *int64   `json:"status,omitempty"`
	Title            *string  `json:"title,omitempty"`
	Type             *string  `json:"type,omitempty"`
	ValidationErrors []string `json:"validation_errors,omitempty"`
}

type PutEventsDeprecateEventID200ApplicationJSON struct {
	Deprecated *string `json:"deprecated,omitempty"`
}

type PutEventsDeprecateEventIDResponse struct {
	ContentType                                       string
	StatusCode                                        int
	PutEventsDeprecateEventID200ApplicationJSONObject *PutEventsDeprecateEventID200ApplicationJSON
	PutEventsDeprecateEventID400ApplicationJSONObject *PutEventsDeprecateEventID400ApplicationJSON
}
